core/payment/src/api/allocations.rs

```diff
@@ -46,44 +112,147 @@ async fn create_allocation(
+                }
+            } else {
+                let network = p.network.as_deref().unwrap_or_else(|| {
+                    if let Some(token) = p.token.as_ref() {
+                        if token == "glm" {
+                            log::debug!("Network not specified, using default polygon, because token set to glm");
+                            "polygon"
+                        } else {
+                            log::debug!("Network not specified, using default holesky");
+                            "holesky"
+                        }
+                    } else {
+                        log::debug!("Network not specified and token not specified, using default holesky");
+                        "holesky"
+                    }
+                });
+                if let Err(err) = validate_network(network) {
+                    let err_msg = format!("Validate network failed (1): {err}");
+                    log::error!("{}", err_msg);
+                    return response::bad_request(&err_msg);
+                }
+                let driver = p.driver.as_deref().unwrap_or_else(|| {
+                    log::debug!("Driver not specified, using default erc20");
+                    "erc20"
+                });
+                if let Err(err) = validate_driver(network, driver) {
+                    let err_msg = format!("Validate driver failed (1): {err}");
+                    log::error!("{}", err_msg);
+                    return response::bad_request(&err_msg);
+                }
+                if let Some(token) = p.token.as_ref() {
+                    if let Err(err) = validate_token(network, driver, token) {
+                        let err_msg = format!("Validate token failed (1): {err}");
+                        log::error!("{}", err_msg);
+                        return response::bad_request(&err_msg);
+                    }
+                    log::debug!("Selected network {driver}-{network}-{token}");
+                    format!("{}-{}-{}", driver, network, token)
+                } else {
+                    let default_token = match get_default_token(driver, network) {
+                        Ok(token) => token,
+                        Err(err) => {
+                            let err_msg = format!("Get default token failed (1): {err}");
+                            log::error!("{}", err_msg);
+                            return response::bad_request(&err_msg);
+                        }
+                    };
+                    log::debug!(
+                        "Selected network with default token {driver}-{network}-{default_token}"
+                    );
+                    format!("{}-{}-{}", driver, network, default_token)
+                }
+            }
+        }
+        None => {
+            log::debug!("No paymentPlatform entry found, using {DEFAULT_PAYMENT_PLATFORM}");
+            DEFAULT_PAYMENT_PLATFORM.to_string()
+        }
     };
```

Comment:
This implementation is so complicated, that you should hide it in separate function